File: docs/examples/nodejs/basic_usage.ts
/**
 * Basic Usage Example - Node.js SDK
 *
 * Copyright (c) 2026 REDSCVRY TECHNOLOGY PRIVATE LIMITED
 * Licensed under the MIT License.
 */

import { Structurify } from '@structurify/sdk';
import { writeFileSync } from 'fs';

async function main() {
  // Initialize client
  const apiKey = process.env.STRUCTURIFY_API_KEY ?? 'sk_live_your_api_key';
  const client = new Structurify({ apiKey });

  // 1. List available templates
  console.log('Available templates:');
  const templates = await client.templates.list();
  for (const template of templates) {
    console.log(`  - ${template.name} (${template.id})`);
  }

  // 2. Create a project from template
  const project = await client.projects.create({
    name: 'Invoice Processing Example',
    templateId: 'tpl_invoice', // Use an actual template ID from step 1
  });
  console.log(`\nCreated project: ${project.id}`);

  // 3. Upload documents
  const documentPaths = ['invoice_001.pdf', 'invoice_002.pdf'];

  for (const docPath of documentPaths) {
    try {
      const doc = await client.documents.uploadFromPath(project.id, docPath);
      console.log(`Uploaded: ${doc.name} (${doc.id})`);
    } catch (error) {
      console.log(`Skipped ${docPath}: file not found`);
    }
  }

  // 4. Run extraction
  console.log('\nStarting extraction...');
  const job = await client.extraction.run(project.id);
  console.log(`Job started: ${job.id}`);

  // 5. Wait for completion
  const completed = await client.extraction.waitForCompletion(job.id, {
    timeout: 300000,
    pollInterval: 2000,
  });
  console.log(
    `Extraction complete: ${completed.completedTasks}/${completed.totalTasks} tasks`
  );

  // 6. Export results
  const exportResult = await client.exports.create({
    projectId: project.id,
    format: 'csv',
  });

  // Check if data is inline or needs download
  let csvData: string;
  if (exportResult.data) {
    csvData = exportResult.data;
  } else {
    csvData = (await client.exports.download(exportResult.export.id)) as string;
  }

  // Save to file
  writeFileSync('export_results.csv', csvData);
  console.log('\nResults exported to export_results.csv');
}

main().catch(console.error);

File: docs/examples/python/basic_usage.py
"""
Basic Usage Example - Python SDK

Copyright (c) 2026 REDSCVRY TECHNOLOGY PRIVATE LIMITED
Licensed under the MIT License.
"""

import os
from structurify import Structurify

# Initialize client
api_key = os.environ.get('STRUCTURIFY_API_KEY', 'sk_live_your_api_key')
client = Structurify(api_key=api_key)

# 1. List available templates
print("Available templates:")
templates = client.templates.list()
for template in templates:
    print(f"  - {template['name']} ({template['id']})")

# 2. Create a project from template
project = client.projects.create(
    name="Invoice Processing Example",
    template_id="tpl_invoice"  # Use an actual template ID from step 1
)
print(f"\nCreated project: {project['id']}")

# 3. Upload documents
documents = [
    "invoice_001.pdf",
    "invoice_002.pdf",
]

for doc_path in documents:
    if os.path.exists(doc_path):
        doc = client.documents.upload(
            project_id=project['id'],
            file_path=doc_path
        )
        print(f"Uploaded: {doc['name']} ({doc['id']})")

# 4. Run extraction
print("\nStarting extraction...")
job = client.extraction.run(project_id=project['id'])
print(f"Job started: {job['id']}")

# 5. Wait for completion
completed = client.extraction.wait_for_completion(
    job['id'],
    timeout=300,
    poll_interval=2.0
)
print(f"Extraction complete: {completed['completedTasks']}/{completed['totalTasks']} tasks")

# 6. Export results
export_result = client.exports.create(
    project_id=project['id'],
    format="csv"
)

# Check if data is inline or needs download
if 'data' in export_result:
    csv_data = export_result['data']
else:
    export_id = export_result.get('export', {}).get('id')
    if export_id:
        csv_data = client.exports.download(export_id)

# Save to file
with open("export_results.csv", "w") as f:
    f.write(csv_data)
print("\nResults exported to export_results.csv")
